### ISSUE TITLE
Hub Menu: Wp Admin url redirects to woocommerce.com

### ISSUE THREAD

**Issue by anitaa1990** [woocommerce/woocommerce-ios#6079]
h/t to @hichamboushaba during beta testing.

```
I noticed after updating the app that the WooCommerce Admin button redirected me to woocommerce.com, I think the field wasn’t present in the Site object after the update, but is it normal to redirect to woocommerce.com in this case? I think it would better to use something like: site_url/wp-admin as a default value
```

cc @pmusolino

**Comment by pmusolino** [comment]
I went deeper into this problem. The reason why we do not redirect to the website address or to wp-admin is that in this case, we have not yet obtained the information to send the user back to the website, so we do not yet know the URL. It should be a rather isolated case, which will happen when there is a network error or during the fetch of site data.

So, we can't redirect to `site_url/wp-admin` because in the case when we redirect to `woocommerce.com` is because we still don't know the `site_url`.
@hichamboushaba do you have any insight on how to replicate the problem? In all my tests, I have not been able to replicate this case.

**Comment by hichamboushaba** [comment]
> So, we can't redirect to `site_url/wp-admin` because in the case when we redirect to `woocommerce.com` is because we still don't know the `site_url`. @hichamboushaba do you have any insight on how to replicate the problem? In all my tests, I have not been able to replicate this case.

I noticed the issue after updating from an old app version, I think it was 8.0 or 8.1 on my device before.

My understanding of the issue is that since we didn't parse and save `admin_url` before (it was added just [recently](https://github.com/woocommerce/woocommerce-ios/pull/5611)), it would be `nil` after updating from an old version, and would be updated only after refetching `/me/sites`.
And regarding `site_url` (or `url` in the model), I don't think it'll be `nil` since it's necessary for the login, it's `adminUrl` which would be `nil` here https://github.com/woocommerce/woocommerce-ios/blob/78d8c49f3a626e4bde9ced117b36a5ec0d1b36b4/WooCommerce/Classes/ViewRelated/Hub%20Menu/HubMenuViewModel.swift#L32-L37

This is a minor issue though 🙂

**Comment by toupper** [comment]
I think this is not relevant anymore. The case @hichamboushaba explained, updating from a version where there was no `adminURL` stored should not happen that often anymore since the admin URL is retrieved since last December.

> I think it would better to use something like: site_url/wp-admin as a default value

We could compose the URL as suggested if we didn't have the admin URL but we did have the site URL, ending up with something like:

```
var woocommerceAdminURL: URL {
        guard let defaultSite = stores.sessionManager.defaultSite else {
            return WooConstants.URLs.blog.asURL()
        }

        guard let url = URL(string: defaultSite.adminURL) else {
            let adminURLFromSiteURLString = defaultSite.url + "\\wp-admin"
            return URL(string: adminURLFromSiteURLString) ?? WooConstants.URLs.blog.asURL()
        }
        return url
    }
```

But I am wondering whether adding this complexity is worth it given what's stated above. What are your thoughts @pmusolino @hichamboushaba ?

**Comment by hichamboushaba** [comment]
A coincidence that I just saw what Jayson shared on this comment here pdND4G-t-p2#comment-43, so I'd say it's better to be fixed since a lot of users are still using old versions.
Another point, the fix would work also if the `admin_url` is `null` in the API response (I don't know if this can happen or not).

But as I said, this is a very minor point, and whatever you decide it would be OK 🙂.

**Comment by itsmeichigo** [comment]
FYI there was [another issue in Jetpack CP](https://github.com/woocommerce/woocommerce-ios/issues/6501) that is pretty similar to this - I had to add a [fallback method](https://github.com/woocommerce/woocommerce-ios/pull/6502) to construct the admin URL manually using the site URL. 

IMO I think it's worth it to add this change - or better yet - maybe add an extension to reuse the logic of fetching/constructing the admin URL. We're using this URL in a few places in the app now.
